File: go/kata/algorithms/binary_search.go
package algorithms

import (
	"math"
)

func BinarySearch(arr []int, target int) bool {
	left := 0
	right := len(arr)

	for left < right {
		mid := int(math.Floor(float64(left + (right-left)/2)))
		value := arr[mid]

		if value == target {
			return true
		}

		if value > target {
			right = mid
		} else {
			left = mid + 1
		}
	}

	return false
}

File: java/kata/src/main/java/br/com/ari/algo/algorithms/BinarySearch.java
package br.com.ari.algo.algorithms;

public class BinarySearch {

    private BinarySearch() {
    }

    public static int search(int[] arr, int target) {
        int left = 0;
        int right = arr.length;

        while (left < right) {
            final int mid = (int) Math.floor(left + (right - left) / 2.0);
            final int value = arr[mid];

            if (value == target) {
                return mid;
            }

            if (value > target) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }

        return -1;
    }

}
